lang/java/reef-experimental/src/main/java/org/apache/reef/experimental/ParquetReader.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+
+import org.apache.parquet.avro.AvroParquetReader;
+import org.apache.parquet.avro.AvroSchemaConverter;
+import org.apache.parquet.format.converter.ParquetMetadataConverter.MetadataFilter;
+import static org.apache.parquet.format.converter.ParquetMetadataConverter.NO_FILTER;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.schema.MessageType;
+
+import javax.inject.Inject;
+
+/**
+ * A reader for Parquet files that can serialize data to local disk and return Avro schema and Avro reader.
+ * The intent is not to build a general parquet reader, but to consume data with table-like property.
+ */
+public class ParquetReader {
+  /**
+   * Standard java logger.
+   */
+  private static final Logger LOG = Logger.getLogger(ParquetReader.class.getName());
+  
+  private Path parquetFilePath;
+
+  @Inject
+  public ParquetReader(final Path path) throws IOException {
+    parquetFilePath = path;
+    Schema schema = getAvroSchema();
+    if (schema.getType() != Schema.Type.RECORD) {
+      LOG.log(Level.SEVERE, "ParquetReader only support Avro record type that can be consumed as a table.");
+    }
+    for (Schema.Field f : schema.getFields()) {
+      if (f.schema().getType() == Schema.Type.RECORD) {
+        LOG.log(Level.SEVERE, "ParquetReader doesn't support nested record type for its elements.");
+      }
+    }
+  }
+  
+  /**
+   * Retrieve avro schema from parquet file.
+   * @return avro schema from parquet file.
+   */
+  public Schema getAvroSchema() throws IOException {
+    return getAvroSchema(new Configuration(true), NO_FILTER);
+  }
+  
+  /**
+   * Retrieve avro schema from parquet file.
+   * @param configuration Hadoop configuration.
+   * @param filter Filter for Avro metadata.
+   * @return avro schema from parquet file.
+   */
+  public Schema getAvroSchema(final Configuration configuration, final MetadataFilter filter) throws IOException {
+    ParquetMetadata footer = ParquetFileReader.readFooter(configuration, parquetFilePath, filter);
+    AvroSchemaConverter converter = new AvroSchemaConverter();
+    MessageType schema = footer.getFileMetaData().getSchema();
+    return converter.convert(schema);
+  }
+  
+  /**
+   * Construct an avro reader from parquet file.
+   * @return avro reader based on the provided parquet file.
+   */
+  public AvroParquetReader<GenericRecord> getAvroReader() throws IOException {
+    return new AvroParquetReader<GenericRecord>(parquetFilePath);
+  }
+  
+  /**
+   * Serialize Avro data to a local file.
+   * @param file Local destination file for serialization.
+   */
+  public void serializeToDisk(final File file) throws IOException {
```

Comment:
Does this need to be `public`? If so, please document the exception with `@throws`